tests/pause_at_block_test.py

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env python3
+
+import json
+import signal
+
+from TestHarness import Account, Cluster, Node, ReturnType, TestHelper, Utils, WalletMgr
+from TestHarness.TestHelper import AppArgs
+
+###############################################################
+# pause_at_block_test
+#
+# Verify /v1/producer/pause_at_block pauses node in all read modes
+#
+###############################################################
+
+# Parse command line arguments
+args = TestHelper.parse_args({"-v","--dump-error-details","--leave-running","--keep-logs","--unshared"})
+Utils.Debug = args.v
+dumpErrorDetails=args.dump_error_details
+dontKill=args.leave_running
+keepLogs=args.keep_logs
+
+walletMgr=WalletMgr(True)
+cluster=Cluster(unshared=args.unshared, keepRunning=args.leave_running, keepLogs=args.keep_logs)
+cluster.setWalletMgr(walletMgr)
+
+testSuccessful = False
+try:
+    TestHelper.printSystemInfo("BEGIN")
+
+    specificNodeosArgs = {
+        0 : "--enable-stale-production",
+
+        3 : "--read-mode head",
+        4 : "--read-mode speculative",
+        5 : "--read-mode irreversible"
+    }
+    assert cluster.launch(
+        pnodes=3,
+        prodCount=3,
+        totalProducers=3,
+        totalNodes=6,
+        loadSystemContract=False,
+        activateIF=True)
+
+    prodNode = cluster.getNode(0)
+    prodNode2 = cluster.getNode(1)
+    headNode = cluster.getNode(3)
+    specNode = cluster.getNode(4)
+    irrvNode = cluster.getNode(5)
+
+    prodNode.waitForProducer("defproducerb");
+    prodNode.waitForProducer("defproducera");
+
+    blockNum = prodNode.getHeadBlockNum()
+
+    blockNum += 5
+
+    Utils.Print(f"Pausing at block {blockNum}")
+    prodNode2.processUrllibRequest("producer", "pause_at_block", {"block_num":blockNum}),
+    headNode.processUrllibRequest("producer", "pause_at_block", {"block_num":blockNum}),
+    specNode.processUrllibRequest("producer", "pause_at_block", {"block_num":blockNum}),
+    irrvNode.processUrllibRequest("producer", "pause_at_block", {"block_num":blockNum}),
+
+    assert prodNode.waitForLibToAdvance(), "LIB did not advance with paused nodes"
+    # blockNum -1 since waitForBlock uses `> blockNum` instead of `>=`
+    assert prodNode2.waitForBlock(blockNum-1), f"Block {blockNum} did not arrive after pausing"
+
+    Utils.Print(f"Verify paused at block {blockNum}")
+    assert prodNode2.getHeadBlockNum() == blockNum, "Prod Node_01 did not pause at block"
+    assert headNode.getHeadBlockNum() == blockNum, "Head Node_02 did not pause at block"
+    assert specNode.getHeadBlockNum() == blockNum, "Speculative Node_03 did not pause at block"
+    assert irrvNode.getHeadBlockNum() == blockNum, "Irreversible Node_04 did not pause at block"
```

Comment:
Regarding the comments, above we have `headNode = cluster.getNode(3)`. Wouldn't that make it `Node_03`?

Reply:
Thanks. I added another node and didn't update the comments.

Reply:
https://github.com/AntelopeIO/spring/pull/1057/commits/836f87d4374b014e1c2e017c828008870af43a13